web/modules/custom/usergreeting/src/GreetingTime.php

```diff
@@ -61,15 +72,18 @@ public function __construct(TimeInterface $timeOutput, AccountInterface $account
    */
   public function greetingMessage(): array {
     $time_output = (int) date('G', $this->timeOutput->getRequestTime());
+    $config_morning = $this->configFactory->get('usergreeting.settings')->get('morning_start');
+    $config_afternoon = $this->configFactory->get('usergreeting.settings')->get('afernoon_start');
+    $config_evening = $this->configFactory->get('usergreeting.settings')->get('evening_start');
 
-    if ($time_output >= 5 && $time_output < 12) {
-      $greeting = $this->t('Good Morning');
+    if ($time_output >= $config_morning && $time_output < $config_afternoon) {
+      $greeting = implode(' ', [$greeting = $this->t('Good Morning')]);
     }
-    elseif ($time_output >= 12 && $time_output < 18) {
-      $greeting = $this->t('Good Afternoon');
+    elseif ($time_output >= $config_afternoon && $time_output < $config_evening) {
+      $greeting = implode(' ', [$greeting = $this->t('Good Afternoon')]);
     }
     else {
-      $greeting = $this->t('Good Evening');
+      $greeting = implode(' ', [$greeting = $this->t('Good Evening')]);
     }
 
     return [
```

Comment:
You could simplify the contents of this render array if you use `#theme => 'username'` only at the outermost depth. Meaning that you don't need to wrap this into a `html_tag`.

Rather than that, you should use `#prefix` and `#suffix` as seen here: https://www.drupal.org/docs/8/api/render-api/render-arrays#properties